core/src/test/java/com/backblaze/b2/json/B2JsonTest.java

```diff
@@ -2147,6 +2147,53 @@ public void testSensitiveWrittenWhenOptionNotSet() {
                 B2Json.toJsonOrThrowRuntime(secureContainer, options));
     }
 
+    private static class OmitNullTestClass {
+        @B2Json.optional(omitNull = true)
+        private final String omitNullString;
+
+        @B2Json.optional
+        private final String regularString;
+
+        @B2Json.optional(omitNull = true)
+        private final int omitNullInt;
+
+        @B2Json.optional
+        private final int regularInt;
+
+        @B2Json.constructor(params = "omitNullString, regularString, omitNullInt, regularInt")
+        public OmitNullTestClass(String omitNullString, String regularString, int omitNullInt, int regularInt) {
+            this.omitNullString = omitNullString;
+            this.regularString = regularString;
+            this.omitNullInt = omitNullInt;
+            this.regularInt = regularInt;
+        }
+    }
+
+    @Test public void testOmitNullWithNullInputs() {
```

Comment:
A test that de-serializes `{}`, then reserializes, might be interesting.

Reply:
implemented